### ISSUE TITLE
Just my code exclusions are cached and can't be updated.

### ISSUE THREAD

**Issue by rchiodo** [microsoft/debugpy#226]
## Environment data

- DEBUGPY version: 1.0.0b8
- OS and version: Windows 10
- Python version (& distribution if applicable, e.g. Anaconda): 3.7.6
- Using VS Code or Visual Studio: VS Code

## Steps to reproduce:
Create a number of files in a folder:
Looper.py:
```
import baz
import foo
while(True):
    inp = input("Enter a string")
    foo.foo()
    baz.baz()
    print('End of loop')
```
foo.py
```
def foo():
    print('foo')
```
baz.py
```
def baz():
    print('baz')
```
And a launch.json with the following (replacing paths appropriately)
```
{
      "name": "Python: Attach using Process Id",
      "type": "python",
      "request": "attach",
      "processId": "${command:pickProcess}",
      "justMyCode": true,
      "rules": [
        {
          "include": true,
          "path": "d:/Training/DebugJMC/foo.py"
        },
        {
          "include": true,
          "path": "d:/Training/DebugJMC/looper.py"
        },
        {
          "include": false,
          "path": "**/*"
        }
      ]
    }
```

1. Start looper.py from VS code
1. Attach and see that breakpoints will bind in foo.py but not in baz.py
1. Detach
1. Switch the include value for foo.py to baz.py
1. Reattach. 

## Actual behavior

Breakpoints don't bind in either

## Expected  behavior

Breakpoints bind in baz.py but not foo.py

**Comment by rchiodo** [comment]
Note: this is necessary for our new run by line feature to work. We need to update the rules on every attach to a different ```<ipython-input-number-hash>``` file.

**Comment by int19h** [comment]
@fabioz, this one is high priority.

**Comment by fabioz** [comment]
I'll take a look at it.

**Comment by fabioz** [comment]
@rchiodo, I've tried to reproduce this with the steps given and I was able to reproduce this with the version shipped with VSCode, but when I set to use the `master` version I couldn't reproduce (and in a test case I also couldn't reproduce), so, it appears that this is fixed with the current master.

i.e.: 
Given that structure with a `baz.py` and a `foo.py` with breakpoints on the functions, when doing the attach to pid with the rules:

```
"rules": [
    {
        "include": true,
        "path": "**/baz.py"
    },
    {
        "include": false,
        "path": "**/*"
    }
]
```

and later with:

```
"rules": [
    {
        "include": true,
        "path": "**/foo.py"
    },
    {
        "include": false,
        "path": "**/*"
    }
]
```

Does seem to disable the breakpoint in the file which is no longer included in the current `master`.

I'll create a pull request with the test I've done in `pydevd` for this (so, I'll leave the issue open for that), but please let me know if you still have some issue with the current `master` version.

**Comment by rchiodo** [comment]
Thanks @fabioz. How do I pull the current master version? Do I just copy the source from where I'm running it?

**Comment by int19h** [comment]
Just clone the repo, and set `"debugAdapterPath": ".../debugpy/adapter"` in debug configuration.

**Comment by rchiodo** [comment]
Our code for loading debugpy actually imports it into the kernel. What path would I force to get that to work? The root of the repo?

**Comment by int19h** [comment]
If you're just doing `import debugpy` etc, then you'll need `src` from the repo to be in `sys.path`.

**Comment by rchiodo** [comment]
It doesn't seem to fix the problem in our run by line scenario. Let me try the straight up python repro

**Comment by rchiodo** [comment]
Yeah it fixes it in the python only scenario. I'll gather debugpy logs.

**Comment by fabioz** [comment]
Ok, waiting for the logs on the actual use case ;)

**Comment by rchiodo** [comment]
Debug logs are pointing it to being my fault :). I wasn't actually updating the config rules when I thought I was with the `<ipython-n-hash>` values. 

Once I fixed that it works now. 

Thanks.

So when the new version ships, our attached [bug ](https://github.com/microsoft/vscode-python/issues/11644) should be fixed.

**Comment by int19h** [comment]
We've already shipped debugpy 1.0.0b9, and bundled it in vscode-python 2020.5.78807
